students/DennisLee/lesson08/circle_class.py

```diff
@@ -0,0 +1,169 @@
+
+        :return:  The resulting summed Circle object.
+        """
+        return self.__add__(other)
+
+    def __mul__(self, other):
+        """
+        Multiply two circles together, with the resulting circle having
+        a radius equal to the product of the individual circle radii.
+
+        :other:  A number or Circle object representing the radius of
+                 the second circle.
+
+        :return:  The resulting multiplied Circle object.
+        """
+        new_radius = self._radius
+        if isinstance(other, (int, float)):
+            new_radius *= other
+        elif isinstance(other, Circle):  
+            new_radius *= other.radius
+        else:
+            raise TypeError(f"Cannot multiply {other} to a circle because it "
+                    f"is a '{type(other)}' type, not an 'int' or 'Circle'.")
+        return Circle(new_radius)
+
+    def __rmul__(self, other):
+        """
+        Multiply two circles together, with the resulting circle having
+        a radius equal to the product of the individual circle radii.
+
+        :other:  A number or Circle object representing the radius of
+                 the first circle.
+
+        :return:  The resulting multiplied Circle object.
+        """
+        return self.__mul__(other)
+
+    def __lt__(self, other):
+        """
+        Return `True` if this circle's radius is less than the other
+        circle's radius.
+        """
+        return self._radius < other.radius
+
+    def __le__(self, other):
+        """
+        Return `True` if this circle's radius is less than or equal to
+        the other circle's radius.
+        """
+        return self._radius <= other.radius
+
+    def __eq__(self, other):
+        """
+        Return `True` if this circle's radius is equal to
+        the other circle's radius.
+        """
+        return self._radius == other.radius
+
+    def __ne__(self, other):
+        """
+        Return `True` if this circle's radius is not equal to
+        the other circle's radius.
+        """
+        return self._radius != other.radius
+
+    def __gt__(self, other):
+        """
+        Return `True` if this circle's radius is greater than the other
+        circle's radius.
+        """
+        return self._radius > other.radius
+
+    def __ge__(self, other):
+        """
+        Return `True` if this circle's radius is greater than or equal to
+        the other circle's radius.
+        """
+        return self._radius >= other.radius
+
+    @property
+    def radius(self):
+        """Get the radius of the circle."""
+        return self._radius
+
+    @radius.setter
+    def radius(self, value):
+        """
+        Set the radius of the circle. Only nonnegative numbers are
+        allowed.
+
+        :value:  The new radius value.
+        """
+        if value < 0:
+            raise ValueError(f"Negative radius of '{value}' not allowed.")
+        else:
+            self._radius = value
```

Comment:
It's great that you're thinking about how to handle faulty user input, but I would use a single instance attribute for `radius` and do the error handling directly in the `__init__` method, rather than using a getter and setter. The approach you're taking ends up giving your class two separate instance attributes for `radius` and `_radius`, which could be a bit confusing for the user. 

It's also worth noting that the leading underscore on `_radius` conventionally denotes a 'protected' attribute (only available for modification in a Class and its subclasses) and I'm not sure if that's what you want to indicate here. Generally, this convention is used fairly sparingly in Python, as it is meant to be an 'open access' language that permits developers to make their own decisions about how they want to use and/or modify their objects.

It's also conventional in Python to define all instance attributes in the `__init__` method, and some linters (like PyCharm's) will complain if you try to define an instance attribute outside of `__init__`, as you're doing for `_radius`. I believe this convention is mainly for code readability, so that other devs can quickly scan the object constructor and know what instance attributes are available.